docs/src/main/asciidoc/security-jpa-concept.adoc

```diff
@@ -0,0 +1,180 @@
+
+
+== JPA entity specification
+
+Quarkus security offers a JPA integration to collect usernames, passwords, and roles, and store them into JPA database entities.
+
+The following JPA entity specification demonstrates how users' information needs to be stored in a JPA entity and properly mapped so that Quarkus can retrieve this information from a database.
+
+
+* The `@UserDefinition` annotation must be present on a JPA entity, no matter if link:https://quarkus.io/guides/hibernate-orm-panache[simplified Hibernate ORM with Panache] is used or not.
+
+* The `@Username` and `@Password` field types are always `String`.
+
+* The `@Roles` field must either be `String`, `Collection<String>`, or a `Collection<X>`, where `X` is an entity class with a single `String` field annotated as `@RolesValue`.
+
+* Each `String` role element type is parsed as a comma-separated list of roles.
+
+The following example demonstrates storing security information by adding annotations to the `user` entity:
+
+[source,java]
+----
+package org.acme.security.jpa;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntity;
+import io.quarkus.elytron.security.common.BcryptUtil;
+import io.quarkus.security.jpa.Password;
+import io.quarkus.security.jpa.Roles;
+import io.quarkus.security.jpa.UserDefinition;
+import io.quarkus.security.jpa.Username;
+
+@Entity
+@Table(name = "test_user")
+@UserDefinition <1>
+public class User extends PanacheEntity {
+    @Username <2>
+    public String username;
+    @Password <3>
+    public String password;
+    @Roles <4>
+    public String role;
+
+    /**
+     * Adds a new user to the database
+     * @param username the username
+     * @param password the unencrypted password (it will be encrypted with bcrypt)
+     * @param role the comma-separated roles
+     */
+    public static void add(String username, String password, String role) { <5>
+        User user = new User();
+        user.username = username;
+        user.password = BcryptUtil.bcryptHash(password);
+        user.role = role;
+        user.persist();
+    }
+}
+
+----
+
+The `security-jpa` extension initializes only if a single entity is annotated with `@UserDefinition`.
+
+<1> The `@UserDefinition` annotation must be present on a single entity, either a regular Hibernate ORM entity or a Hibernate ORM with a Panache entity.
+<2> Indicates the field used for the username.
+<3> Indicates the field used for the password.
+By default, it uses bcrypt-hashed passwords, and you can configure it to use plain text or custom passwords.
+<4> This indicates the comma-separated list of roles added to the target principal representation attributes.
+<5> This method allows us to add users while hashing passwords with the proper `bcrypt` hash.
```

Comment:
Suggestion:Change _**us**_ to **_you_** to be consistent with previous statements.